inugami_webapp/src/main/webapp/js/app/components/forms/input.time.slots.ts

```diff
@@ -0,0 +1,233 @@
+
+    @Input() labelOn                : string;
+
+    @Input() labelOff               : string;
+
+    @Input() validator              : any;
+
+    @Output() onFocus               : EventEmitter<any> = new EventEmitter();
+
+    @Output() onChange              : EventEmitter<any> = new EventEmitter();
+
+    @Output() onAdd                 : EventEmitter<any> = new EventEmitter();
+
+    @Output() onDelete              : EventEmitter<any> = new EventEmitter();
+
+    onModelChange                   : Function = () => { };
+
+    onModelTouched                  : Function = () => { };
+
+    styleClasses                    : string[];
+
+    focused                         : boolean;
+
+    valid                           : boolean = true;
+
+    enableValidator                 : boolean = false;
+
+    timeSlotsModel                  : any[] = [];
+
+    separator                       : string;
+
+    
+
+    /*****************************************************************************
+    * INIT
+    *****************************************************************************/
+
+    ngOnInit() {
+       if(this.timeSlotsModel.length == 0){
+         let initialTimeSlot = new TimeSlotBuilder().withFrom("00:00")
+                                                    .withTo("23:59")
+                                                    .build();
+         this.timeSlotsModel.push(initialTimeSlot);
+         this.separator= org.inugami.formatters.message("time.slots.to");
+       }
+    }
+    constructor() { }
+
+    /****************** ***********************************************************
+     * IMPLEMENTS ControlValueAccessor
+     *****************************************************************************/
+    writeValue(timeSlots: any): void {
+        this.timeSlotsModel = timeSlots;
+    }
+
+    registerOnChange(fn: Function): void {
+        this.onModelChange = fn;
+    }
+
+    registerOnTouched(fn: Function): void {
+        this.onModelTouched = fn;
+    }
+    setDisabledState(val: boolean): void {
+        this.disabled = val;
+    }
+
+    /*****************************************************************************
+    * METHODS
+    *****************************************************************************/
+
+    addSlot(from = "00:00", to = "23:59"){
+      if(!this.readonly && !this.disabled){
+        let newSlot = new TimeSlotBuilder().withFrom(from)
+                                          .withTo(to)
+                                          .build();
+        this.timeSlotsModel.push(newSlot);
+        let addEventResponse = {"added":newSlot,
+                                "data":this.timeSlotsModel} 
+        this.onAdd.emit(addEventResponse);
+      }
+    }
+    removeSlot(index : number){
+      if(!this.readonly && !this.disabled){
+        let removedSlot = this.timeSlotsModel[index];
+        this.timeSlotsModel.splice(index,1);
+
+        let removeEventResponse = {"deleted":removedSlot,
```

Comment:
When you fire an event, you should create a model for better integration of your component in other view or plugins